sdks/python/apache_beam/runners/direct/sdf_direct_runner.py

```diff
@@ -173,26 +201,64 @@ def process(self, element, timestamp=beam.DoFn.TimestampParam,
       state.add_state(window, self._element_tag, element)
       state.add_state(window, self._restriction_tag,
                       sdf_result.residual_restriction)
+      # Holding output watermark by setting it to negative infinity.
       state.add_state(window, self.watermark_hold_tag,
                       WatermarkManager.WATERMARK_NEG_INF)
-      # Holding output watermark by setting it to negative infinity.
+
+      # Setting a timer to be reinvoked to continue processing the element.
+      # Currently Python SDK only supports setting timers based on watermark. So
+      # forcing a reinvocation by setting a timer for watermark negative
+      # infinity.
```

Comment:
Does this have any practical consequences, except that the processing-time delay is not respected? E.g. does this unnecessarily hold the watermark more than requested, or something? Can the timer end up being dropped?